src/omnisharp/loggingEvents.ts

```diff
@@ -21,78 +25,90 @@ export class OmnisharpDelayTrackerEventMeasures extends TelemetryEventWithMeasur
 export class OmnisharpStart extends TelemetryEventWithMeasures {
 }
 
-export class OmnisharpInitialisation implements BaseEvent{
+export class OmnisharpInitialisation implements BaseEvent {
     constructor(public timeStamp: Date, public solutionPath: string) { }
 }
 
-export class OmnisharpLaunch implements BaseEvent{
+export class OmnisharpLaunch implements BaseEvent {
     constructor(public usingMono: boolean, public command: string, public pid: number) { }
 }
 
-export class PackageInstallation implements BaseEvent{
+export class PackageInstallation implements BaseEvent {
     constructor(public packageInfo: string) { }
 }
 
-export class LogPlatformInfo implements BaseEvent{
+export class LogPlatformInfo implements BaseEvent {
     constructor(public info: PlatformInformation) { }
 }
 
-export class InstallationProgress implements BaseEvent{
+export class InstallationProgress implements BaseEvent {
     constructor(public stage: string, public message: string) { }
 }
 
-export class InstallationFailure implements BaseEvent{
-    constructor(public stage: string,public error: any) { }
+export class InstallationFailure implements BaseEvent {
+    constructor(public stage: string, public error: any) { }
 }
 
-export class DownloadProgress implements BaseEvent{
+export class DownloadProgress implements BaseEvent {
     constructor(public downloadPercentage: number) { }
 }
 
-export class OmnisharpFailure implements BaseEvent{
+export class OmnisharpFailure implements BaseEvent {
     constructor(public message: string, public error: Error) { }
 }
 
-export class OmnisharpRequestMessage implements BaseEvent{
+export class OmnisharpRequestMessage implements BaseEvent {
     constructor(public request: Request, public id: number) { }
 }
 
-export class TestExecutionCountReport implements BaseEvent{
+export class TestExecutionCountReport implements BaseEvent {
     constructor(public debugCounts: { [testFrameworkName: string]: number }, public runCounts: { [testFrameworkName: string]: number }) { }
 }
 
-export class OmnisharpServerOnError implements BaseEvent{
+export class OmnisharpServerOnError implements BaseEvent {
     constructor(public errorMessage: protocol.ErrorMessage) { }
 }
 
-export class OmnisharpServerMsBuildProjectDiagnostics implements BaseEvent{
+export class OmnisharpServerMsBuildProjectDiagnostics implements BaseEvent {
     constructor(public diagnostics: protocol.MSBuildProjectDiagnostics) { }
 }
 
-export class OmnisharpServerUnresolvedDependencies implements BaseEvent{
-    constructor(public unresolvedDependencies: protocol.UnresolvedDependenciesMessage) { }
+export class OmnisharpServerUnresolvedDependencies implements BaseEvent {
+    constructor(public unresolvedDependencies: protocol.UnresolvedDependenciesMessage, public server: OmniSharpServer, public eventStream: EventStream) { }
```

Comment:
This change is temporary, right? We'll get rid of the extra parameters soon?

Reply:
Yes we will use the vscode api's to do the restore then we can get rid of this here.